docs/async/async-csharp-vb.rst

```diff
@@ -0,0 +1,200 @@
+
+Tasks are awaitable, meaning that the application of the ``await`` keyword will allow for either:
+
+* "Unwrapping" the return value for a ``Task<T>`` operation after it completes.
+* Simply waiting for a ``Task`` operation to finish before moving forward.
+
+Tasks are also used outside of the async programming model.  They are the foundation of the Task Parallel Library, which allows support for `Data Parallelism <https://msdn.microsoft.com/en-us/library/dd537608(v=vs.110).aspx>`_ and `Task Parallelism <https://msdn.microsoft.com/en-us/library/dd537609(v=vs.110).aspx>`_.  However, from the perspective of writing responsive code with the C#/VB async programming model, ``Task`` and ``Task<T>`` are really just used as a means to coordinating blocking operations and extracting their return values (if they have them).
+
+Important Info and Advice
+-------------------------
+
+Although async programming is relatively straightfoward, there are some details to keep in mind which could otherwise result in some nasty behavior.
+
+* **Best practice is to add "Async" to the end of every async method name you write which could be consumed by another method.**
+
+Failure to do so could result in having to track down a race condition later.  It's better to be explicit here!  Note that certain methods which aren't explicity called by your code (such as event handlers or web controller methods) may not necessarily apply.
+
+* ``await`` **is what will ultimately make a method asynchronous.**
+
+Failing to apply the ``await`` operator to a task will result in the async method running synchronously!  Application of ``await`` is what suspends the async method, giving back control to the method which called it.  Pay attention to compiler warnings about this.
+
+* ``async void`` **should only be used for event handlers.**
+
+Why?  That's the only reason they were allowed in the first place!  Async programming uses the ``Task`` and ``Task<T>`` objects, which provide flexibility in coordinating asynchronous work.  Throwing that out of the window with ``async void`` doesn't follow the model very well.  Here's some specific issues:
+
+    (a) Exceptions thrown in an ``async void`` method can't be caught outside of that method.
+	
+    (b) ``async void`` methods are very difficult to test.
+	
+    (c) ``async void`` methods can cause bad side effects if the caller isn't expecting them to be async.
+	
+That said, ``async void`` is perfect for event handlers where the event involves any blocking task(s).
+
+* **Tread carefully when using async lambdas in LINQ expressions**
```

Comment:
This is begging for a link to a document that provides more detail and guidance.